Add getCacheId() and clearCache() method to Translator

MatthiasKuehneEllerhold opened the pull request:
See https://github.com/zendframework/zend-i18n/pull/63 for discussion.

mlocati committed:
Add getCacheId method to Translator

MatthiasKuehneEllerhold committed:
Make getCacheId() a public function and add clearCache()

malukenho commented on src/Translator/Translator.php:
```diff
@@ -561,6 +561,34 @@ public function addRemoteTranslations($type, $textDomain = 'default')
+     * Get the cache identifier for a specific textDomain and locale.
+     *
+     * @param    string $textDomain
+     * @param    string $locale
```
Too much space

malukenho commented on src/Translator/Translator.php:
```diff
@@ -561,6 +561,34 @@ public function addRemoteTranslations($type, $textDomain = 'default')
+     * Clears the cache for a specific textDomain and locale.
+     *
+     * @param $textDomain
+     * @param $locale
```
Parameter types?

malukenho reviewed (changes requested):
I’m missing tests for the introduced api

MatthiasKuehneEllerhold committed:
Style Fixes + Unit-Tests

MatthiasKuehneEllerhold commented:
Fixed the comments and added tests.

Is there any consensus in the ZF code style about using static methods via `$this` ? (See Unit-Tests `$this->assertXY`...

weierophinney commented:
> Is there any consensus in the ZF code style about using static methods via $this ?

For unit tests, that's the style we've consistently done since, well, we began. There's an [open discussion in our forums covering exactly this](https://discourse.zendframework.com/t/this-assert-vs-self-assert/448), though with no consensus as of yet.

weierophinney commented:
Thanks, @MatthiasKuehneEllerhold. Merged to develop for release with 2.9.0.
